dev/plugins/hu.elte.txtuml.xtxtuml/src/hu/elte/txtuml/xtxtuml/formatting2/XtxtUMLFormatter.xtend

```diff
@@ -137,8 +138,14 @@ class XtxtUMLFormatter extends XbaseFormatter {
 		}
 	}
 
+	def dispatch void format(TUModifiers it, extension IFormattableDocument document) {
+		regionFor.feature(TU_MODIFIERS__STATIC).append[oneSpace];
+		regionFor.feature(TU_MODIFIERS__EXTERNALITY).append[oneSpace];
+		regionFor.feature(TU_MODIFIERS__VISIBILITY).append[oneSpace];	
+	}
+	
 	def dispatch void format(TUConstructor it, extension IFormattableDocument document) {
-		regionFor.feature(TU_CONSTRUCTOR__VISIBILITY).append[oneSpace];
```

Comment:
If this is not possible in case of an attribute or operation declaration prefix, I think we should not do this in this case either. Also, I do not like at all the idea to do something twice only to preserve some invariant which we cannot really preserve. As I see it, it is a lot worse than simply admitting the failure to preserve the invariant and not just because it reduces performance but because it is a lie! If someone reads the code and finds no explicit information about this violation, they simply assume that the invariant is preserved and add changes to the code that result in surprising failures.

 I think the only clean solution here is to add an explanatory comment here.